src/Microsoft.Dnx.Runtime/Servicing/ServicingTable.cs

```diff
@@ -21,51 +21,28 @@ public static class ServicingTable
             string assetPath,
             out string replacementPath)
         {
-            var compositeIndex = LoadIndex();
-
-            foreach (var index in compositeIndex)
-            {
-                if (index.TryGetReplacement(packageId, packageVersion, assetPath, out replacementPath))
-                {
-                    return true;
-                }
-            }
-
-            replacementPath = null;
-            return false;
+            return LoadIndex().TryGetReplacement(packageId, packageVersion, assetPath, out replacementPath);
         }
 
-        private static List<ServicingIndex> LoadIndex()
+        private static ServicingIndex LoadIndex()
         {
             return LazyInitializer.EnsureInitialized(ref _index, ref _indexInitialized, ref _indexSync, () =>
             {
-                var compositeIndex = new List<ServicingIndex>();
-
-                foreach (var servicingRoot in GetServicingRoots())
+                var index = new ServicingIndex();
+                var servicingRoot = GetServicingRoot();
+                if (servicingRoot != null && servicingRoot.IndexOfAny(Path.GetInvalidPathChars()) == -1)
```

Comment:
It was added to address this comment: https://github.com/aspnet/dnx/pull/2183#discussion_r33820594. However with the current changes it may no longer be necessary since dnx will fail if it can't read the index.txt file and will never actually get here.